src/js/components/Shared/TracksList.react.js

```diff
@@ -189,158 +124,247 @@ export default class TracksList extends Component {
+                        onMouseDown={ self.selectTrack }
+                        onContextMenu={ self.showContextMenu }
+                    >
+                        <div className='cell cell-track-playing text-center'>
+                            { playingIndicator }
+                        </div>
+                        <div className='cell cell-track'>
+                            { track.title }
+                        </div>
+                        <div className='cell cell-duration'>
+                            { utils.parseDuration(track.duration) }
+                        </div>
+                        <div className='cell cell-artist'>
+                            { track.artist[0] }
+                        </div>
+                        <div className='cell cell-album'>
+                            { track.album }
+                        </div>
+                        <div className='cell cell-genre'>
+                            { track.genre.join(', ') }
+                        </div>
+                    </TrackRow>
+                );
+            });
+
+            const translationDistance = (tilesScrolled * this.rowHeight * chunkLength) +
+                                        (indexChunk * this.rowHeight * chunkLength);
+            const tracksListTileStyles = {
+                transform: `translate3d(0, ${translationDistance}px, 0)`
+            };
+
+            return (
+                <div className='tracks-list-tile' key={ indexChunk } style={ tracksListTileStyles }>
+                    { list }
+                </div>
+            );
+        });
+    }
+
+    _pausePlayState() {
+        Player.getAudio().paused ? 'pause' : 'play';
+    }
+
+    _toggleSelectionById(id) {
+        let selected = [...this.state.selected];
+
+        if(selected.includes(id)) {
+            // remove track
+            selected.splice(selected.indexOf(id), 1);
+
+        } else {
+            // add track
+            selected.push(id);
+        }
+
+        selected = utils.simpleSort(selected, 'asc');
+        this.setState({ selected });
+    }
+
+    _multiSelect(e, id, index) {
+        const self   = this;
+        const tracks = this.props.tracks;
+        const selected = this.state.selected;
+
+        switch(selected.length) {
+            case 0: {
+                selected.push(id);
+                this.setState({ selected });
                 break;
+            }
+            case 1: {
+
+                const onlySelected = selected[0];
+                let onlySelectedIndex;
+
+                // for(let i = 0, length = tracks.length; i < length; i++) { // STH wrong here
+                //     if(tracks[i]._id === onlySelected) {
+                //         onlySelectedIndex = i;
+                //         break;
+                //     }
+                // }
+                const onlySelectedIndex = tracks.findIndex((track) => return track._id === onlySelected);
```

Comment:
Use  `bla.map((elem) => elem.thing)` for return statements in fat arrow function, but not in plain function with braces (like `isRightClick: function() {}`)